tests/test-war-smoke/src/test/java/com/google/cloud/runtimes/jetty/test/smoke/ForwardedIntegrationTest.java

```diff
@@ -0,0 +1,144 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.runtimes.jetty.test.smoke;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.google.cloud.runtime.jetty.test.AbstractIntegrationTest;
+import com.google.cloud.runtime.jetty.test.annotation.LocalOnly;
+import com.google.cloud.runtime.jetty.test.annotation.RemoteOnly;
+import com.google.cloud.runtime.jetty.util.HttpUrlUtil;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+
+public class ForwardedIntegrationTest extends AbstractIntegrationTest {
+
+  /**
+   * Validate proxy proto header is handled.
+   *
+   * @throws IOException test in error
+   */
+  @Test
+  @LocalOnly
+  public void testLocalProto() throws IOException {
+    URI target = getUri().resolve("/dump/info");
+    assertThat(target.getPath(), containsString("/dump/info"));
+
+    HttpURLConnection http = HttpUrlUtil.openTo(target);
+    http.setRequestProperty("X-Forwarded-Proto", "https");
+
+    assertThat(http.getResponseCode(), is(200));
+    String responseBody = HttpUrlUtil.getResponseBody(http);
+    assertThat(responseBody, containsString("scheme=https(secure=true)"));
+  }
+  
+  /**
+   * Validate proxy for header is handled.
+   *
+   * @throws IOException test in error
+   */
+  @Test
+  @LocalOnly
+  public void testLocalFor() throws IOException {
+    URI target = getUri().resolve("/dump/info");
+    assertThat(target.getPath(), containsString("/dump/info"));
+
+    HttpURLConnection http = HttpUrlUtil.openTo(target);
+    http.setRequestProperty("X-Forwarded-For", "1.2.3.4,5.6.7.8");
+
+    assertThat(http.getResponseCode(), is(200));
+    String responseBody = HttpUrlUtil.getResponseBody(http);
+    assertThat(responseBody, containsString("remoteHost/Addr:port=1.2.3.4"));
+  }
+
+  /**
+   * Validate proxy proto header is handled.
+   *
+   * @throws IOException test in error
+   */
+  @Test
+  @RemoteOnly
+  public void testRemoteProto() throws IOException {
+    URI target = getUri().resolve("/dump/info");
+    assertThat(target.getPath(), containsString("/dump/info"));
+
+    HttpURLConnection http = HttpUrlUtil.openTo(target);
+    http.setRequestProperty("X-Forwarded-Proto", "https");
+
+    assertThat(http.getResponseCode(), is(200));
+    String responseBody = HttpUrlUtil.getResponseBody(http);
+    
+    // TODO should we trust the client to say it is secure?  Not secure to the GFE!
+    assertThat(responseBody, containsString("scheme=https(secure=true)"));
```

Comment:
I'm not following you?  The default scheme for `getUri()` is not `https`.  It is only `https` in jetty if something has customized the request be be so, of which there is only the secureRequestCustomizer used on real SSL connections and the HttpForwardedCustomizer that we are using here process the X-Forwarded-Proto header.

You can see this on the URIs:
 - http://20161223t162320-dot-jetty9-work.appspot-preview.com/dump
 - https://20161223t162320-dot-jetty9-work.appspot-preview.com/dump

I